lib/diagnostic.js

```diff
@@ -4,32 +4,68 @@
 // Describe in your own words what a `callback`
 // is in relation to array methods
 
+They are basically a function that store value to be passed inside another function
+
```

Comment:
A callback is a function passed to an array method to do whatever is specified in the callback on each array element